antarest/study/dao/link/composite_link_dao.py

```diff
@@ -0,0 +1,57 @@
+# Copyright (c) 2024, RTE (https://www.rte-france.com)
+#
+# See AUTHORS.txt
+#
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+#
+# SPDX-License-Identifier: MPL-2.0
+#
+# This file is part of the Antares project.
+
+import typing as t
+
+from antarest.core.interfaces.cache import ICache
+from antarest.study.business.model.link_model import LinkBaseDTO, LinkDTO
+from antarest.study.dao.link.cache_link_dao import CacheLinkDao
+from antarest.study.dao.link.storage_link_dao import StorageLinkDao
+from antarest.study.model import Study
+from antarest.study.storage.storage_service import StudyStorageService
+
+
+class CompositeLinkDAO:
```

Comment:
See other comment:
I think the composite DAO et the storage DAO should both implement the same interface (abstract class), so they can be used interchangeably.

The cache "DAO" is better separated because actually it's not really a DAO (it has different methods).